modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java

```diff
@@ -673,7 +673,12 @@ public void processOpts() {
             additionalProperties.put(RESPONSE_WRAPPER, "CompletableFuture");
         }
         if (reactive) {
-            additionalProperties.put(RESPONSE_WRAPPER, "Mono");
```

Comment:
Not sure if this is right. I'am not a reactive expert, but here looks like you can use ResponseEntity together with reactive in different ways.

https://docs.spring.io/spring-framework/reference/web/webflux/controller/ann-methods/responseentity.html

Reply:
Indeed and in fact my patch is only about the situation where reactive is activated and responseEntity is deactivated.
For example, the example "springboot-reactive" where both reactive and responseEntity are activated is left untouched.
For example, the operation `findPetsByStatus`:

|  | Reactive enabled | Reactive disabled |
|--------|--------|--------|
| ResponseEntity enabled | `Mono<ResponseEntity<Flux<Pet>>>` | `ResponseEntity<List<Pet>>` |
| ResponseEntity disabled | `Flux<Pet>` | `List<Pet>` |

Extracted from the samples:
* springboot-delegate
* springboot-delegate-no-response-entity
* springboot-reactive
* springboot-reactive-noResponseEntity

And as you can see in the modified files, my patch only has impact on the sample `springboot-reactive-noResponseEntity`